src/execution_plan/ops/op_delete.c

```diff
@@ -189,50 +186,76 @@ static Record DeleteConsume(OpBase *opBase) {
 
 			SIValue_Free(value);
 		} else if(!(type & T_NULL)) {
-			/* Expression evaluated to a non-graph entity type
-			 * clear pending deletions and raise an exception. */
-			array_clear(op->deleted_nodes);
-			array_clear(op->deleted_edges);
-			// If evaluating the expression allocated any memory, free it.
+			// if evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
 			// free the Record this operation acted on
 			OpBase_DeleteRecord(r);
```

Comment:
I think this is problematic, because this operation is eager and because an exception is raised we must free all collected records within `DeleteFree` at the moment this isn't done...